common/app/html/HtmlLinkUtmInsertion.scala

```diff
@@ -0,0 +1,35 @@
+package html
+
+import java.net.URL
+import common.Logging
+import model.EmailAddons
+import org.jsoup.Jsoup
+import org.jsoup.nodes._
+import play.twirl.api.Html
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object HtmlLinkUtmInsertion extends Logging {
+
+  def apply(html: Html): Html = {
+    val documentBody = Jsoup.parse(html.toString)
+    Html(setLinks(documentBody).toString)
+  }
+
+  private def setLinks(element: Element): Element = {
+    Option(element.attr("href"))
+      .collect { case url if url.nonEmpty && element.nodeName() == "a" && !url.contains(EmailAddons.unsubscribePlaceholder) =>
```

Comment:
Rather than excluding the unsubscribe link, how about only including www, or excluding https://profile.* ?

Reply:
I chose not to implement this since I don't think Braze's future implementation will be able to do this.

It would be necessary to parse the URLs and only apply the placeholder to ones without the hostname specified, or a host beginning with `www.`

Reply:
👍